lib/crawler/api/config.rb

```diff
@@ -330,12 +332,29 @@ def configure_domain_allowlist!
       end
 
       def validate_domain!(domain)
-        url = URI.parse(domain)
-        raise ArgumentError, "Domain #{domain.inspect} does not have a URL scheme" unless url.scheme
-        raise ArgumentError, "Domain #{domain.inspect} is not an HTTP(S) site" unless url.is_a?(URI::HTTP)
+        url = Addressable::URI.parse(domain)
```

Comment:
I guess Adressable::URI.parse supports more stuff than URI.parse?

Reply:
Basically yes. In-built `URI` lib doesn't handle IDNs at all, but the addressable gem does.
We were already using addressable elsewhere in the codebase actually, just missed it for validation here.